app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -151,8 +151,15 @@ def get_instances
         process_collection(instances, :vms) { |instance| parse_instance(instance) }
       end
 
+      # The underlying method that gathers these images is a bit brittle.
+      # Consequently, if it raises an error we just log it and move on so
+      # that it doesn't affect the rest of inventory collection.
+      #
       def get_images
         images = gather_data_for_this_region(@sas, 'list_all_private_images')
+      rescue Azure::Armrest::ApiException => err
+        _log.warn("Unable to collect Azure private images: #{err.message}")
```

Comment:
Include `@ems` data here, like the `@ems.name` ... or anything (region, subscription) that will help the user correlate this warning to the specific EMS.